src/checkpoint/buffer/io_buffer.cc

```diff
@@ -117,20 +182,20 @@ void IOBuffer::setupFile() {
   debug_checkpoint("IOBuffer: mmap file: len=%lu\n", size_);
 
   /*
-   * mmap or mmap64 the file descriptor in WRITE mode, PRIVATE to this process
+   * mmap or mmap64 the file descriptor in WRITE mode
    */
   void* addr = nullptr;
 
 # if defined(checkpoint_has_mmap64)
-  addr = mmap64(nullptr, size_, PROT_WRITE, MAP_PRIVATE, fd_, 0);
+  addr = mmap64(nullptr, size_, PROT_WRITE, MAP_SHARED, fd_, 0);
 
   if (addr == MAP_FAILED) {
     auto err = std::string("mmap64 failed for writing file: errno=") +
                std::to_string(errno);
     throw std::runtime_error(err);
   }
 # else
-  addr = mmap(nullptr, size_, PROT_WRITE, MAP_PRIVATE, fd_, 0);
+  addr = mmap(nullptr, size_, PROT_WRITE, MAP_SHARED, fd_, 0);
```

Comment:
I'm a little confused how `MAP_PRIVATE` was a defect in the write path - when should multiple processes have a shared mapping of a file that should be unique per process?

Reply:
Yes, it was writing a all-zero file with `MAP_PRIVATE`. Not sure why.